src/fromager/resolver.py

```diff
@@ -200,12 +206,14 @@ def __init__(
         include_wheels: bool = True,
         sdist_server_url: str = "https://pypi.org/simple/",
         constraints: Constraints | None = None,
+        req_type: RequirementType | None = None,
```

Comment:
It feels a little odd to have the requirement type passed to the constructor instead of validate_candidate(). That implies the resolver will only resolve this type of dependencies. What led to that choice?

How much more code has to change if we add the argument to validate_candidate()?

Reply:
validate_candidate is called by find_matches which is internally called by the resolvelib library functions. I am not sure how we will pass req_type to validate_candidate

Reply:
Ah, ok, it sounds like we can't.